From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Alexander <protonull@protonmail.com>
Date: Mon, 12 Jul 2021 08:50:38 +0100
Subject: [PATCH] Add PhantomPreSpawnClusterEvent


diff --git a/src/main/java/io/protonull/deepstate/event/entity/PhantomPreSpawnClusterEvent.java b/src/main/java/io/protonull/deepstate/event/entity/PhantomPreSpawnClusterEvent.java
new file mode 100644
index 0000000000000000000000000000000000000000..f2e885a3e6c52235faa45b104d7752080f24c9f1
--- /dev/null
+++ b/src/main/java/io/protonull/deepstate/event/entity/PhantomPreSpawnClusterEvent.java
@@ -0,0 +1,51 @@
+package io.protonull.deepstate.event.entity;
+
+import org.bukkit.entity.Player;
+import org.bukkit.event.HandlerList;
+import org.bukkit.event.player.PlayerEvent;
+import org.jetbrains.annotations.NotNull;
+
+public class PhantomPreSpawnClusterEvent extends PlayerEvent {
+	private static final HandlerList handlers = new HandlerList();
+
+	private int clusterCount;
+	private final int timeSinceRest;
+
+	public PhantomPreSpawnClusterEvent(@NotNull final Player target, final int clusterCount, final int timeSinceRest) {
+		super(target);
+		this.clusterCount = clusterCount;
+		this.timeSinceRest = timeSinceRest;
+	}
+
+	/**
+	 * @return Returns how many phantoms will be spawned on the target player.
+	 */
+	public int getClusterCount() {
+		return Math.max(this.clusterCount, 0);
+	}
+
+	/**
+	 * @param clusterCount Sets how many phantoms will be spawned on the target player.
+	 */
+	public void setClusterCount(final int clusterCount) {
+		this.clusterCount = clusterCount;
+	}
+
+	/**
+	 * @return Returns the target player's time since rest statistic's value.
+	 */
+	public int getTimeSinceRest() {
+		return this.timeSinceRest;
+	}
+
+	@NotNull
+	@Override
+	public HandlerList getHandlers() {
+		return handlers;
+	}
+
+	@NotNull
+	public static HandlerList getHandlerList() {
+		return handlers;
+	}
+}
