From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Alexander <protonull@protonmail.com>
Date: Fri, 16 Jul 2021 21:22:02 +0100
Subject: [PATCH] Add world UUID caching to Location


diff --git a/src/main/java/org/bukkit/Location.java b/src/main/java/org/bukkit/Location.java
index c9b953aa55eeee87e81b9b712c0f501a58e641fb..15731642089e8ccfa1e3332a8672e7706ddd88a8 100644
--- a/src/main/java/org/bukkit/Location.java
+++ b/src/main/java/org/bukkit/Location.java
@@ -22,6 +22,10 @@ import org.bukkit.entity.LivingEntity;
 import org.bukkit.entity.Player;
 // Paper end
 
+// DeepState start
+import java.util.UUID;
+// DeepState end
+
 /**
  * Represents a 3-dimensional position in a world.
  * <br>
@@ -32,6 +36,7 @@ import org.bukkit.entity.Player;
  */
 public class Location implements Cloneable, ConfigurationSerializable {
     private Reference<World> world;
+    private UUID worldUUID; // DeepState
     private double x;
     private double y;
     private double z;
@@ -61,9 +66,14 @@ public class Location implements Cloneable, ConfigurationSerializable {
      * @param pitch The absolute rotation on the y-plane, in degrees
      */
     public Location(@UndefinedNullability final World world, final double x, final double y, final double z, final float yaw, final float pitch) {
+		// DeepState start
+    	/*
         if (world != null) {
             this.world = new WeakReference<>(world);
         }
+         */
+		setWorld(world);
+		// DeepState end
 
         this.x = x;
         this.y = y;
@@ -79,6 +89,7 @@ public class Location implements Cloneable, ConfigurationSerializable {
      */
     public void setWorld(@Nullable World world) {
         this.world = (world == null) ? null : new WeakReference<>(world);
+        this.worldUUID = world == null ? null : world.getUID(); // DeepState
     }
 
     /**
@@ -113,6 +124,17 @@ public class Location implements Cloneable, ConfigurationSerializable {
         return world;
     }
 
+    // DeeState start
+    /**
+	 * @return Returns this world's uuid, which can be null. Please do not construe a non-null value as implying that
+	 *         the world is loaded, use {@link #isWorldLoaded()} for that information.
+	 */
+    @Nullable
+    public UUID getWorldUUID() {
+    	return this.worldUUID;
+	}
+	// DeeState end
+
     /**
      * Gets the chunk at the represented location
      *
