From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Alexander <protonull@protonmail.com>
Date: Sat, 17 Jul 2021 11:30:43 +0100
Subject: [PATCH] Cache full NamespacedKey


diff --git a/src/main/java/org/bukkit/NamespacedKey.java b/src/main/java/org/bukkit/NamespacedKey.java
index 01bcb3a1bdb5accdf844d0178cec3d25746b3eaa..acf9ca2cd936b9651f297b60f0d9af7aef385941 100644
--- a/src/main/java/org/bukkit/NamespacedKey.java
+++ b/src/main/java/org/bukkit/NamespacedKey.java
@@ -37,6 +37,8 @@ public final class NamespacedKey implements net.kyori.adventure.key.Key, com.des
     private final String namespace;
     private final String key;
 
+    private final String fullKeyCache; // DeepState add key cache
+
     /**
      * Create a key in a specific namespace.
      *
@@ -52,6 +54,8 @@ public final class NamespacedKey implements net.kyori.adventure.key.Key, com.des
         this.namespace = namespace;
         this.key = key;
 
+        this.fullKeyCache = this.namespace + ':' + this.key; // DeepState add key cache
+
         String string = toString();
         Preconditions.checkArgument(string.length() < 256, "NamespacedKey must be less than 256 characters", string);
     }
@@ -75,6 +79,8 @@ public final class NamespacedKey implements net.kyori.adventure.key.Key, com.des
         this.namespace = plugin.getName().toLowerCase(Locale.ROOT);
         this.key = key.toLowerCase(Locale.ROOT);
 
+        this.fullKeyCache = this.namespace + ':' + this.key; // DeepState add key cache
+
         // Check validity after normalization
         Preconditions.checkArgument(VALID_NAMESPACE.matcher(this.namespace).matches(), "Invalid namespace. Must be [a-z0-9._-]: %s", this.namespace);
         Preconditions.checkArgument(VALID_KEY.matcher(this.key).matches(), "Invalid key. Must be [a-z0-9/._-]: %s", this.key);
@@ -117,7 +123,7 @@ public final class NamespacedKey implements net.kyori.adventure.key.Key, com.des
 
     @Override
     public String toString() {
-        return this.namespace + ":" + this.key;
+        return this.fullKeyCache; //this.namespace + ":" + this.key; // DeepState
     }
 
     /**
@@ -231,7 +237,7 @@ public final class NamespacedKey implements net.kyori.adventure.key.Key, com.des
     @NotNull
     @Override
     public String asString() {
-        return this.namespace + ':' + this.key;
+        return this.fullKeyCache; //this.namespace + ":" + this.key; // DeepState
     }
     // Paper end
 }
