From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Alexander <protonull@protonmail.com>
Date: Sat, 17 Jul 2021 12:32:57 +0100
Subject: [PATCH] Add conditional block getter to Location


diff --git a/src/main/java/org/bukkit/Location.java b/src/main/java/org/bukkit/Location.java
index 2e0b8421246e591dad9f3efcecc49bc38a5577a7..0116587f691872abcb004924d4ebbfd3bee8c32d 100644
--- a/src/main/java/org/bukkit/Location.java
+++ b/src/main/java/org/bukkit/Location.java
@@ -155,6 +155,26 @@ public class Location implements Cloneable, ConfigurationSerializable {
         return getWorld().getBlockAt(this);
     }
 
+    // DeepState start
+    @Nullable
+    public Block getBlockIfLoaded() {
+        return getBlockIfLoaded(false);
+    }
+    @Nullable
+    public Block getBlockIfLoaded(final boolean checkWorldLoaded) {
+        if (checkWorldLoaded && !isWorldLoaded()) {
+            return null;
+        }
+        final var world = getWorld();
+        final var blockX = locToBlock(this.x);
+        final var blockZ = locToBlock(this.z);
+        if (!world.isChunkLoaded(blockX >> 4, blockZ >> 4)) {
+            return null;
+        }
+        return world.getBlockAt(blockX, locToBlock(this.y), blockZ);
+    }
+    // DeepState end
+
     /**
      * Sets the x-coordinate of this location
      *
