From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Alexander <protonull@protonmail.com>
Date: Sun, 27 Jun 2021 22:24:48 +0100
Subject: [PATCH] Compilation fix


diff --git a/build.gradle.kts b/build.gradle.kts
index 8ccb5183af4a10a92d17570833e21dfffb5b03ea..124424cf437ca50609c13ad8ac3ef1dff068799e 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -3,9 +3,9 @@ import com.github.jengelman.gradle.plugins.shadow.transformers.Transformer
 import io.papermc.paperweight.tasks.BaseTask
 import io.papermc.paperweight.tasks.GenerateReobfMappings
 import io.papermc.paperweight.tasks.PatchMappings
-import io.papermc.paperweight.util.Constants
 import io.papermc.paperweight.util.Git
 import io.papermc.paperweight.util.cache
+import io.papermc.paperweight.util.constants.*
 import io.papermc.paperweight.util.defaultOutput
 import io.papermc.paperweight.util.openZip
 import io.papermc.paperweight.util.path
@@ -36,7 +36,7 @@ repositories {
 }
 
 dependencies {
-    implementation(project(":Tuinity-API")) // Tuinity
+    implementation(project(":DeepState-API")) // DeepState
     implementation("com.destroystokyo.paper:paper-mojangapi:1.16.5-R0.1-SNAPSHOT") // Tuinity
     // Paper start
     implementation("org.jline:jline-terminal-jansi:3.12.1")
@@ -118,8 +118,8 @@ val patchReobfMappings by tasks.registering<PatchMappings> {
     inputMappings.set(generateReobfMappings.flatMap { it.reobfMappings })
     patch.set(rootProject.layout.cache.resolve("paperweight/upstreams/paper/build-data/reobf-mappings-patch.tiny"))
 
-    fromNamespace.set(Constants.DEOBF_NAMESPACE)
-    toNamespace.set(Constants.SPIGOT_NAMESPACE)
+    fromNamespace.set(DEOBF_NAMESPACE)
+    toNamespace.set(SPIGOT_NAMESPACE)
 
     outputMappings.set(layout.cache.resolve("paperweight/mappings/reobf-patched.tiny"))
 }
