From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Alexander <protonull@protonmail.com>
Date: Sun, 27 Jun 2021 22:24:48 +0100
Subject: [PATCH] Compilation fix


diff --git a/build.gradle.kts b/build.gradle.kts
index ab8de6c4e3c0bea2b9f498da00adf88e987d2364..80096c0ad2920a8c4ea6f6c963a5eea5c9cc6d00 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -36,7 +36,7 @@ repositories {
 }
 
 dependencies {
-    implementation(project(":Tuinity-API")) // Tuinity
+    implementation(project(":DeepState-API")) // DeepState
     implementation("com.destroystokyo.paper:paper-mojangapi:1.16.5-R0.1-SNAPSHOT") // Tuinity
     // Paper start
     implementation("org.jline:jline-terminal-jansi:3.12.1")
