From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Alexander <protonull@protonmail.com>
Date: Sun, 27 Jun 2021 23:31:50 +0100
Subject: [PATCH] Allow GameProfile name re-setting


diff --git a/src/main/java/com/mojang/authlib/GameProfile.java b/src/main/java/com/mojang/authlib/GameProfile.java
index 1369f0e680bd581a7fb93fb37ef5ab3d6aee5c98..4dbfe62290a13295ab26d717ff0c99854261f3a0 100644
--- a/src/main/java/com/mojang/authlib/GameProfile.java
+++ b/src/main/java/com/mojang/authlib/GameProfile.java
@@ -8,7 +8,7 @@ import java.util.UUID;
 
 public class GameProfile {
     private final UUID id;
-    private final String name;
+    private String name;
     private final PropertyMap properties = new PropertyMap();
     private boolean legacy;
 
@@ -52,6 +52,15 @@ public class GameProfile {
         return name;
     }
 
+    /**
+     * Re-sets the name for this game profile. Doing this should be avoided unless absolutely necessary.
+     *
+     * @param name The name to set onto this game profile.
+     */
+    public void UNSAFE_setName(final String name) {
+        this.name = name;
+    }
+
     /**
      * Returns any known properties about this game profile.
      *
