From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Alexander <protonull@protonmail.com>
Date: Sun, 27 Jun 2021 23:38:08 +0100
Subject: [PATCH] Emit BeaconTickEvent


diff --git a/src/main/java/net/minecraft/world/level/block/entity/BeaconBlockEntity.java b/src/main/java/net/minecraft/world/level/block/entity/BeaconBlockEntity.java
index c1a0b0d77b8783fd127b68449a209ec0e62e6005..2dfbe5f23737293c0037cb5b058be79b7c6ed6a9 100644
--- a/src/main/java/net/minecraft/world/level/block/entity/BeaconBlockEntity.java
+++ b/src/main/java/net/minecraft/world/level/block/entity/BeaconBlockEntity.java
@@ -204,6 +204,8 @@ public class BeaconBlockEntity extends BlockEntity implements MenuProvider {
             if (blockEntity.levels > 0 && !blockEntity.beamSections.isEmpty()) {
                 BeaconBlockEntity.applyEffects(world, pos, blockEntity.levels, blockEntity.primaryPower, blockEntity.secondaryPower);
                 BeaconBlockEntity.playSound(world, pos, SoundEvents.BEACON_AMBIENT);
+
+                new io.protonull.deepstate.event.block.BeaconTickEvent(org.bukkit.craftbukkit.block.CraftBlock.at(world, pos)).callEvent();
             }
         }
         // Paper start - beacon activation/deactivation events
