From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Alexander <protonull@protonmail.com>
Date: Mon, 12 Jul 2021 08:51:36 +0100
Subject: [PATCH] Emit PhantomPreSpawnClusterEvent to make cluster size
 changeable


diff --git a/src/main/java/net/minecraft/world/level/levelgen/PhantomSpawner.java b/src/main/java/net/minecraft/world/level/levelgen/PhantomSpawner.java
index 79504dc3448402e73b09c4232b1fd0488872cf68..deca721e52f0ad4f1b13f176461091fcd8ae070c 100644
--- a/src/main/java/net/minecraft/world/level/levelgen/PhantomSpawner.java
+++ b/src/main/java/net/minecraft/world/level/levelgen/PhantomSpawner.java
@@ -73,6 +73,10 @@ public class PhantomSpawner implements CustomSpawner {
                                             SpawnGroupData groupdataentity = null;
                                             int k = 1 + random.nextInt(difficultydamagescaler.getDifficulty().getId() + 1);
 
+                                            final var clusterEvent = new io.protonull.deepstate.event.entity.PhantomPreSpawnClusterEvent(((ServerPlayer) entityhuman).getBukkitEntity(), k);
+                                            clusterEvent.callEvent();
+                                            k = clusterEvent.getClusterCount();
+
                                             for (int l = 0; l < k; ++l) {
                                                 // Paper start
                                                 com.destroystokyo.paper.event.entity.PhantomPreSpawnEvent event = new com.destroystokyo.paper.event.entity.PhantomPreSpawnEvent(MCUtil.toLocation(world, blockposition1), ((ServerPlayer) entityhuman).getBukkitEntity(), org.bukkit.event.entity.CreatureSpawnEvent.SpawnReason.NATURAL);
