From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Alexander <protonull@protonmail.com>
Date: Fri, 16 Jul 2021 21:09:19 +0100
Subject: [PATCH] Add public constructor to CraftItemStack


diff --git a/src/main/java/org/bukkit/craftbukkit/inventory/CraftItemStack.java b/src/main/java/org/bukkit/craftbukkit/inventory/CraftItemStack.java
index b3acc23d54c593b599517a481ad6ac86e5661fa2..0f1d4b7a2ecddf197f70273bf548d5c986a434fd 100644
--- a/src/main/java/org/bukkit/craftbukkit/inventory/CraftItemStack.java
+++ b/src/main/java/org/bukkit/craftbukkit/inventory/CraftItemStack.java
@@ -17,6 +17,11 @@ import org.bukkit.enchantments.Enchantment;
 import org.bukkit.inventory.ItemStack;
 import org.bukkit.inventory.meta.ItemMeta;
 import org.bukkit.material.MaterialData;
+// DeepState start
+import java.util.Objects;
+import javax.annotation.Nonnull;
+import org.bukkit.Bukkit;
+// DeepState end
 
 @DelegateDeserialization(ItemStack.class)
 public final class CraftItemStack extends ItemStack {
@@ -103,6 +108,12 @@ public final class CraftItemStack extends ItemStack {
         this.setItemMeta(itemMeta);
     }
 
+    // DeepState start
+    public CraftItemStack(@Nonnull final Material type, final int amount) {
+        this(Objects.requireNonNull(type), amount, (short) 0, Bukkit.getItemFactory().getItemMeta(type));
+    }
+    // DeepState end
+
     @Override
     public MaterialData getData() {
         return this.handle != null ? CraftMagicNumbers.getMaterialData(this.handle.getItem()) : super.getData();
