From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Alexander <protonull@protonmail.com>
Date: Sat, 17 Jul 2021 01:05:42 +0100
Subject: [PATCH] Add CraftItemStack versions of ItemStack utility methods


diff --git a/src/main/java/org/bukkit/craftbukkit/inventory/CraftItemStack.java b/src/main/java/org/bukkit/craftbukkit/inventory/CraftItemStack.java
index 0f1d4b7a2ecddf197f70273bf548d5c986a434fd..123d52f153b5e0146c0d212b6c66ef3395325593 100644
--- a/src/main/java/org/bukkit/craftbukkit/inventory/CraftItemStack.java
+++ b/src/main/java/org/bukkit/craftbukkit/inventory/CraftItemStack.java
@@ -18,9 +18,13 @@ import org.bukkit.inventory.ItemStack;
 import org.bukkit.inventory.meta.ItemMeta;
 import org.bukkit.material.MaterialData;
 // DeepState start
+import io.protonull.deepstate.utilities.ClosableHandler;
 import java.util.Objects;
 import javax.annotation.Nonnull;
 import org.bukkit.Bukkit;
+import org.bukkit.craftbukkit.persistence.CraftPersistentDataContainer;
+import org.bukkit.craftbukkit.persistence.CraftPersistentDataTypeRegistry;
+import org.bukkit.persistence.PersistentDataContainer;
 // DeepState end
 
 @DelegateDeserialization(ItemStack.class)
@@ -591,4 +595,26 @@ public final class CraftItemStack extends ItemStack {
     static boolean hasItemMeta(net.minecraft.world.item.ItemStack item) {
         return !(item == null || item.getTag() == null || item.getTag().isEmpty());
     }
+
+    // DeepState start
+    public ClosableHandler<PersistentDataContainer> getPersistentDataContainer() {
+        final var nbt = this.handle.getOrCreateTag();
+        final var raw = nbt.getCompound(BUKKIT_CUSTOM_TAG.NBT).copy();
+        final var pdc = new CraftPersistentDataContainer(raw.tags, new CraftPersistentDataTypeRegistry());
+        final var initialRawHash = raw.hashCode();
+        return new ClosableHandler<>() {
+            @Override
+            public PersistentDataContainer get() {
+                return pdc;
+            }
+            @Override
+            public void close() {
+                if (pdc.hashCode() != initialRawHash) {
+                    nbt.put(BUKKIT_CUSTOM_TAG.NBT, raw);
+                }
+            }
+        };
+    }
+    // DeepState end
+
 }
